python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -874,6 +879,76 @@ def _update_dist_lookup_table_vars(self, param_list, grad_list,
                 ]
         return param_list, grad_list
 
+    def _append_distributed_metric(self):
+        support_distributed_metric_ops = ['auc']
+        self.distributed_metric_ops = []
+        for op in self.origin_program.global_block().ops:
+            if op.type in support_distributed_metric_ops and op.attr(
+                    'is_distributed') is True:
+                self.distributed_metric_ops.append(op)
+
+        if len(self.distributed_metric_ops) == 0:
+            self.has_distributed_metrics = False
+            return
+
+        self.has_distributed_metrics = True
+
+        for op in self.distributed_metric_ops:
+            if op.type == "auc":
+                self._append_distributed_auc(op)
+
+    def _append_distributed_auc(self, auc_op):
```

Comment:
can we not use a transpiler? like using some code like:

server: 
```python
with fluid.layers.ListenAndServe():
    fluid.layers.auc()
```

clients:
```python
...
fluid.layers.send()
```